From e5658c60414627502ed1e9aea77a3086d53e100a Mon Sep 17 00:00:00 2001
From: Rosen Penev <rosenp@gmail.com>
Date: Thu, 30 Jul 2026 12:00:29 -0700
Subject: [PATCH 12531/15333] crypto: cesa - clear cesa_dev on _remove

This is already done on probe failure.

cesa_dev should be set to NULL as the driver checks that it is not and
errors saying that only one device is authorized.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
---
 drivers/crypto/marvell/cesa/cesa.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/crypto/marvell/cesa/cesa.c b/drivers/crypto/marvell/cesa/cesa.c
index 4859ad2e86b4..564b09773507 100644
--- a/drivers/crypto/marvell/cesa/cesa.c
+++ b/drivers/crypto/marvell/cesa/cesa.c
@@ -544,6 +544,8 @@ static void mv_cesa_remove(struct platform_device *pdev)
 	struct mv_cesa_dev *cesa = platform_get_drvdata(pdev);
 
 	mv_cesa_remove_algs(cesa);
+
+	cesa_dev = NULL;
 }
 
 static const struct platform_device_id mv_cesa_plat_id_table[] = {
-- 
2.55.0

